app/validators/end_date_after_start_date_validator.rb

```diff
@@ -0,0 +1,13 @@
+class EndDateAfterStartDateValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return unless value
+
+    unless end_date_after_start_date(record.planned_start_date, record.planned_end_date)
+      record.errors.add :planned_end_date, I18n.t("activerecord.errors.models.activity.attributes.planned_end_date.not_before_start_date")
```

Comment:
You could shorten this, as the validator already knows where to look in the I18n scopes:

```ruby
record.errors.add :planned_end_date, :not_before_start_date
```